core/src/main/java/feast/core/service/AccessManagementService.java

```diff
@@ -42,11 +42,16 @@
   private AuthorizationProvider authorizationProvider;
   private ProjectRepository projectRepository;
 
+  // TODO: Remove duplication of constructor
```

Comment:
@dr3s is it possible to remove the `ObjectProvider` here to remove the code duplication in two nearly identical `AccessManagementService()` constructors? This makes the code really error prone. 
(:disappointed_relieved: I was stuck debugging because I didn't realize there were two constructors.)
It seems that one constructor is used by Spring and one used by tests.